image-hd.c

```diff
@@ -236,7 +245,9 @@ static int hdimage_insert_ebr(struct image *image, struct partition *part)
 		entry->partition_type = 0x0F;
 		entry->relative_sectors = (p->offset - hd->align - hd->extended_lba)/512;
 		entry->total_sectors = (p->size + hd->align)/512;
```

Comment:
Unrelated to your change, so no need to change it here, but I'm pretty sure this `total_sectors` calculation is wrong. From reading Wikipedia, this should be the size of the _next_ logical partition (including ebr and alignment)...

Reply:
I surprised that sfdisk doesn't change that as well.

Reply:
Hmm, good catch, might be worth looking into as well. But I just double-checked with sfdisk there's now really just the difference in the partition type, with this layout:
```
Device      Boot Start   End Sectors Size Id Type
sfdisk.img1       2048  4095    2048   1M 83 Linux
sfdisk.img2       4096 28671   24576  12M  f W95 Ext'd (LBA)
sfdisk.img5       6144 10239    4096   2M 83 Linux
sfdisk.img6      12288 14335    2048   1M 83 Linux
sfdisk.img7      16384 20479    4096   2M 83 Linux
sfdisk.img8      22528 24575    2048   1M 83 Linux
sfdisk.img9      26624 28671    2048   1M 83 Linux
```
the diff of sfdisk and genimage images looks like this:
```diff
--- /dev/fd/63	2024-06-03 11:49:01.772013580 +0200
+++ /dev/fd/62	2024-06-03 11:49:01.776013569 +0200
@@ -4,19 +4,19 @@
 000001f0: 0000 0000 0000 0000 0000 0000 0000 55aa  ..............U.
 002001b0: 0000 0000 0000 0000 0000 0000 0000 0061  ...............a
 002001c0: 2200 83a2 2200 0008 0000 0010 0000 00a2  "..."...........
-002001d0: 2300 05e3 2300 0018 0000 0010 0000 0000  #...#...........
+002001d0: 2300 0fe3 2300 0018 0000 0010 0000 0000  #...#...........
 002001f0: 0000 0000 0000 0000 0000 0000 0000 55aa  ..............U.
 005001b0: 0000 0000 0000 0000 0000 0000 0000 00c3  ................
 005001c0: 0400 83e3 2300 0008 0000 0008 0000 00e3  ....#...........
-005001d0: 2400 0546 0501 0028 0000 0018 0000 0000  $..F...(........
+005001d0: 2400 0f46 0501 0028 0000 0018 0000 0000  $..F...(........
 005001f0: 0000 0000 0000 0000 0000 0000 0000 55aa  ..............U.
 007001b0: 0000 0000 0000 0000 0000 0000 0000 0005  ................
 007001c0: 0501 8346 0501 0008 0000 0010 0000 0046  ...F...........F
-007001d0: 0601 0587 0601 0040 0000 0010 0000 0000  .......@........
+007001d0: 0601 0f87 0601 0040 0000 0010 0000 0000  .......@........
 007001f0: 0000 0000 0000 0000 0000 0000 0000 55aa  ..............U.
 00a001b0: 0000 0000 0000 0000 0000 0000 0000 0066  ...............f
 00a001c0: 2601 8387 0601 0008 0000 0008 0000 0087  &...............
-00a001d0: 0701 05c8 0701 0050 0000 0010 0000 0000  .......P........
+00a001d0: 0701 0fc8 0701 0050 0000 0010 0000 0000  .......P........
 00a001f0: 0000 0000 0000 0000 0000 0000 0000 55aa  ..............U.
 00c001b0: 0000 0000 0000 0000 0000 0000 0000 00a7  ................
 00c001c0: 2701 83c8 0701 0008 0000 0008 0000 0000  '...............
```